### PR TITLE
Add SecurityScope property to Device object

### PR DESCRIPTION
**Pull request by varunpuranik**
## Checklist
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [X] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.
This PR is against securityscopes-preview branch.

## Description of the changes
Add SecurityScope property to Device Object
Update Api-version to the one IoTHub expects

### CODE REVIEW

File: iothub/service/src/Device.cs [hunk 1]
```diff
@@ -108,5 +108,11 @@ public string ETag
         /// </summary>
         [JsonProperty(PropertyName = "capabilities", NullValueHandling = NullValueHandling.Ignore)]
         public virtual DeviceCapabilities Capabilities { get; set; }
+
+        /// <summary>
+        /// Scope to which this device instance belongs to
+        /// </summary>
+        [JsonProperty(PropertyName = "securityScope", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (CIPop) [inline iothub/service/src/Device.cs]: @varunpuranik please add an E2E test to verify this actually works.

File: e2e/test/IoTHubServiceProxyE2ETests.cs [hunk 2]
```diff
@@ -46,6 +46,27 @@ public async Task RegistryManager_AddAndRemoveDevice_WithProxy()
             await RegistryManager_AddDevice(httpTransportSettings).ConfigureAwait(false);
         }
 
+        [TestMethod]
```
Comment (CIPop) [inline e2e/test/IoTHubServiceProxyE2ETests.cs]: @varunpuranik  please create a separate file for RegistryManager tests (we didn't have this before). 
This file is for HTTP proxy tests only.

Comment (varunpuranik) [reply e2e/test/IoTHubServiceProxyE2ETests.cs]: Sure

**Comment by CIPop** [comment]
Thanks @varunpuranik for adding the test. (@myagley FYI)

1. The hub fails all E2E tests because the version is not yet recognized. We need to defer the merge until we have the new preview version environment ready.
2. Please squash after addressing the test location issue.

**Comment by CIPop** [comment]
We have an environment available now. Trying to merge.

**Pull request merged**